src/main/java/org/sagebionetworks/bridge/spring/controllers/UploadController.java

```diff
@@ -175,6 +175,10 @@ public UploadView getUpload(@PathVariable String uploadId) {
             if (record == null) {
                 throw new EntityNotFoundException(HealthDataRecord.class);
             }
+            // You cannot authenticate in one study, and then retrieve a record from another study
+            if (!session.getStudyIdentifier().getIdentifier().equals(record.getStudyId())) {
+                throw new UnauthorizedException("Study admin cannot retrieve upload in another study.");
+            }
```

Comment:
It's very common for me to have an upload ID but not know which study it belongs to. I think it would be better to use verifyCrossAdminStudy() instead.

Also, the Worker also calls this API, and this would kill the Worker.